hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azure/AzureNativeFileSystemStore.java

```diff
@@ -1858,7 +1895,25 @@ public void storeEmptyLinkFile(String key, String tempBlobKey,
       CloudBlobWrapper blob = getBlobReference(key);
       storePermissionStatus(blob, permissionStatus);
       storeLinkAttribute(blob, tempBlobKey);
-      openOutputStream(blob).close();
+      if (eTagCheck && eTag != null) {
+        AccessCondition accessCondition = new AccessCondition();
+        accessCondition.setIfMatch(eTag);
```

Comment:
I think this logic is incorrect. This method is creating an empty link file but we are creating the empty file conditionally if it matches the `eTag` of the original file.

The logic for `create` should be the following (pseudocode):
```
eTag = null
if (overwrite) {
  eTag = readETag(dst)
  storeEmptyLinkFile(tmp)
  ...
  commitTmpFileTo(tmp, dst, eTag)  // should do IfMatch(eTag)
} else {
  storeEmptyLinkFile(tmp)
  ...
  commitTmpFileTo(tmp, dst, null)  // should do IfNoneMatch("*")
}
```
and in all cases, inside `storeEmptyLinkFile` we should use `IfNoneMatch("*")` because we know we do not want to overwrite the tmp file and we guarantee that it is unique by putting a UUID in its name.

Reply:
The storeEmptyLinkFile call is made for the original file and not the temp file. Hence if it is case of file overwrite it will have if-match eTag condition else if-none-match *. The check for temp file is done in storeFile method in AzureNativeFileSystemStore class

Reply:
In line 1895 the blob being referenced is for the original key